pinot-core/src/main/java/org/apache/pinot/core/indexsegment/mutable/MutableSegmentImpl.java

```diff
@@ -432,202 +409,193 @@ public void addExtraColumns(Schema newSchema) {
+      String column = entry.getKey();
+      IndexContainer indexContainer = entry.getValue();
       Object value = row.getValue(column);
-      NumValuesInfo numValuesInfo = _numValuesInfoMap.get(column);
+      FieldSpec fieldSpec = indexContainer._fieldSpec;
       if (fieldSpec.isSingleValueField()) {
-        // SV column
-        MutableForwardIndex mutableForwardIndex = _forwardIndexMap.get(column);
-        Integer dictId = (Integer) dictIdMap.get(column);
-        if (dictId != null) {
-          // SV Column with dictionary
-          mutableForwardIndex.setDictId(docId, dictId);
+        // Single-value column
+
+        // Update numValues info
+        indexContainer._numValuesInfo.updateSVEntry();
+
+        // Update indexes
+        MutableForwardIndex forwardIndex = indexContainer._forwardIndex;
+        int dictId = indexContainer._dictId;
+        if (dictId >= 0) {
+          // Dictionary-encoded single-value column
+
+          // Update forward index
+          forwardIndex.setDictId(docId, dictId);
+
+          // Update inverted index
+          RealtimeInvertedIndexReader invertedIndex = indexContainer._invertedIndex;
+          if (invertedIndex != null) {
+            invertedIndex.add(dictId, docId);
+          }
         } else {
-          // No-dictionary SV column
+          // Single-value column with raw index
+
+          // Update forward index
           DataType dataType = fieldSpec.getDataType();
           switch (dataType) {
             case INT:
-              mutableForwardIndex.setInt(docId, (Integer) value);
+              forwardIndex.setInt(docId, (Integer) value);
               break;
             case LONG:
-              mutableForwardIndex.setLong(docId, (Long) value);
+              forwardIndex.setLong(docId, (Long) value);
               break;
             case FLOAT:
-              mutableForwardIndex.setFloat(docId, (Float) value);
+              forwardIndex.setFloat(docId, (Float) value);
               break;
             case DOUBLE:
-              mutableForwardIndex.setDouble(docId, (Double) value);
+              forwardIndex.setDouble(docId, (Double) value);
               break;
             case STRING:
-              mutableForwardIndex.setString(docId, (String) value);
+              forwardIndex.setString(docId, (String) value);
               break;
             case BYTES:
-              mutableForwardIndex.setBytes(docId, (byte[]) value);
+              forwardIndex.setBytes(docId, (byte[]) value);
               break;
             default:
               throw new UnsupportedOperationException(
                   "Unsupported data type: " + dataType + " for no-dictionary column: " + column);
           }
+
+          // Update min/max value from raw value
+          // NOTE: Skip updating min/max value for aggregated metrics because the value will change over time.
+          if (!_aggregateMetrics || fieldSpec.getFieldType() != FieldSpec.FieldType.METRIC) {
```

Comment:
Is the check for aggregateMetrics redundant here?

Reply:
No, without `aggregateMetrics` we can store min/max value for metric columns because the value won't change over time